tests/test_generics.py

```diff
@@ -1704,6 +1704,23 @@ class Model2(BaseModel, Generic[S]):
     Model2[str].model_rebuild()
 
 
+def test_generic_recursive_models_parametrized_with_model() -> None:
+    """https://github.com/pydantic/pydantic/issues/11748"""
+
+    T = TypeVar('T')
+
+    class Base(BaseModel, Generic[T]):
+        t: T
+
+    class Other(BaseModel):
+        child: 'Base[Other]'
+
+    with pytest.raises(ValidationError):
+        # In v2.0-2.10, this validated fine (The core schema of Base[Other].t was an empty model).
+        # Since v2.11, building `Other` raised an unhandled exception.
+        Base[Other].model_validate({'t': {}})
```

Comment:
Can we also add an example that does validate? Just to make sure that we didn't (and won't in the future) accidentally break this pattern entirely.